app/controllers/users_controller.rb

```diff
@@ -0,0 +1,16 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.all
+  end
+
+  def show
+    @user = User.find(params[:id])
+    @posts = @user.posts
+  end
+
+  def posts
+    @user = User.find(params[:id])
+    @posts = @user.posts
+    @user_name = @user.name
+  end
```

Comment:
- [ ] Please remove this action `posts` from here, It will be covered in the `posts#index` action as I requested above.